start.sh

```diff
@@ -0,0 +1,243 @@
+    ${COMPOSE} pull
+}
+
+# get the IP:port of a container via either the local docker-machine or from
+# sdc-listmachines.
+getIpPort() {
+    if [ -z "${COMPOSE_CFG}" ]; then
+        local ip=$(sdc-listmachines --name ${PREFIX}_$1_1 | json -a ips.1)
+        local port=$2
+    else
+        local ip=$(docker-machine ip default)
+        local port=$(docker inspect ${PREFIX}_$1_1 | json -a NetworkSettings.Ports."$2/tcp".0.HostPort)
+    fi
+    echo "$ip:$port"
+}
+
+# start and initialize the Couchbase cluster, along with Consul
+startDatabase() {
+    echo
+    echo 'Starting Couchbase'
+    ${COMPOSE} up -d --no-recreate couchbase
+    echo
+    echo -n 'Initializing cluster.'
+
+    sleep 1.3
+    COUCHBASERESPONSIVE=0
+    while [ $COUCHBASERESPONSIVE != 1 ]; do
+        echo -n '.'
+        RUNNING=$(docker inspect "${PREFIX}_couchbase_1" | json -a State.Running)
+        if [ "$RUNNING" == "true" ]
+        then
+            docker exec -it "${PREFIX}_couchbase_1" triton-bootstrap bootstrap benchmark
+            let COUCHBASERESPONSIVE=1
+        else
+            sleep 1.3
+        fi
+    done
+}
+
+# open the web consoles
+showConsoles() {
+    local CONSUL=$(getIpPort consul 8500)
+    echo
+    echo 'Consul is now running'
+    echo "Dashboard: $CONSUL"
+    command -v open >/dev/null 2>&1 && `open http://${CONSUL}/ui/`
+
+    local CBDASHBOARD=$(getIpPort couchbase 8091)
+    echo
+    echo 'Couchbase cluster running and bootstrapped'
+    echo "Dashboard: $CBDASHBOARD"
+    echo 'The username and password are printed in earlier messages'
+    command -v open >/dev/null 2>&1 && `open http://${CBDASHBOARD}/index.html#sec=servers`
+}
+
+# send a REST API call to remove a CB bucket
+removeBucket() {
+    curl -X DELETE -vvv http://${CBAPI}/pools/default/buckets/$1 \
+         -u ${CB_USER}:${CB_PASSWORD}
+}
+
+# use Docker exec to use the Couchbase CLI to create a CB bucket;
+# this avoids using the REST API so we don't have to deal with proxy port
+# configuration
+createBucket() {
+    docker exec -it ${PREFIX}_couchbase_1 \
+           /opt/couchbase/bin/couchbase-cli bucket-create -c 127.0.0.1:8091 \
+           -u ${CB_USER} -p ${CB_PASSWORD} \
+           --bucket=$1 \
+           --bucket-type=couchbase \
+           --bucket-ramsize=${CB_RAM_QUOTA} \
+           --bucket-replica=1 \
+           --wait # need to wait otherwise index creation can fail later
+}
+
+# send a REST API call to Couchbase to create a N1QL index
+createIndex() {
+    echo $1
+    curl -s --fail -s -X POST http://${N1QLAPI}/query/service \
+         -u ${CB_USER}:${CB_PASSWORD} \
+         -d "statement=$1"
+}
+
+# create all buckets and indexes we need. if you modify the names
+# of the buckets in config.json you'll need to modify this section
+setupCouchbase() {
+    CBAPI=$(getIpPort couchbase 8091)
+    N1QLAPI=$(getIpPort couchbase 8093)
+    echo
+    echo 'Creating Couchbase buckets'
+    removeBucket benchmark
```

Comment:
I get a failure on this semi-regularly, though it might be the bad wifi:

```
Creating Couchbase buckets
*   Trying 165.225.159.37...
* Connected to 165.225.159.37 (165.225.159.37) port 8091 (#0)
* Server auth using Basic with user 'Administrator'
> DELETE /pools/default/buckets/benchmark HTTP/1.1
> Host: 165.225.159.37:8091
> Authorization: Basic QWRtaW5pc3RyYXRvcjpwYXNzd29yZA==
> User-Agent: curl/7.43.0
> Accept: */*
> 
< HTTP/1.1 404 Object Not Found
< Server: Couchbase Server
< Pragma: no-cache
< Date: Mon, 16 Nov 2015 10:01:50 GMT
< Content-Type: text/plain
< Content-Length: 31
< Cache-Control: no-cache
< 
Requested resource not found.
* Connection #0 to host 165.225.159.37 left intact
..SUCCESS: bucket-create
...SUCCESS: bucket-create
HTTP/1.1 500 Connection Not Upgraded
Connection: close
Date: Mon, 16 Nov 2015 10:02:35 GMT

Creating Couchbase indexes
CREATE PRIMARY INDEX ON users
```